packages/client/hmi-client/src/page/data-explorer/components/selected-resources-header-pane.vue

```diff
@@ -0,0 +1,180 @@
+
+const router = useRouter();
+
+const props = defineProps({
+	selectedSearchItems: {
+		type: Array as PropType<ResultType[]>,
+		required: true
+	}
+});
+
+const emit = defineEmits(['close']);
+const resources = useResourcesStore();
+
+const validProject = computed(() => resources.activeProject);
+
+const projectsList = ref<Project[]>([]);
+const projectsNames = computed(() => projectsList.value.map((p) => p.name));
+
+const addResourcesToProject = async (projectId: string) => {
+	// send selected items to the store
+	props.selectedSearchItems.forEach(async (selectedItem) => {
+		if (isDocument(selectedItem)) {
+			const body: DocumentAsset = {
+				xdd_uri: (selectedItem as DocumentType).gddId,
+				title: (selectedItem as DocumentType).title
+			};
+
+			// FIXME: handle cases where assets is already added to the project
+
+			// first, insert into the proper table/collection
+			const res = await addDocuments(body);
+			if (res) {
+				const documentId = res.id;
+
+				// then, link and store in the project assets
+				const assetsType = ProjectAssetTypes.DOCUMENTS;
+				await ProjectService.addAsset(projectId, assetsType, documentId);
+
+				// update local copy of project assets
+				validProject.value?.assets?.[ProjectAssetTypes.DOCUMENTS].push(documentId);
+				resources.activeProjectAssets?.[ProjectAssetTypes.DOCUMENTS].push(body);
+			}
+		}
+		if (isModel(selectedItem)) {
+			// FIXME: handle cases where assets is already added to the project
+			const modelId = selectedItem.id;
+			// then, link and store in the project assets
+			const assetsType = ProjectAssetTypes.MODELS;
+			await ProjectService.addAsset(projectId, assetsType, modelId);
+
+			// update local copy of project assets
+			validProject.value?.assets.models.push(modelId);
+			resources.activeProjectAssets?.[ProjectAssetTypes.MODELS].push(selectedItem);
+		}
+		if (isDataset(selectedItem)) {
+			// FIXME: handle cases where assets is already added to the project
+			const datasetId = selectedItem.id;
+			// then, link and store in the project assets
+			const assetsType = ProjectAssetTypes.DATASETS;
+			await ProjectService.addAsset(projectId, assetsType, datasetId);
+
+			// update local copy of project assets
+			validProject.value?.assets.datasets.push(datasetId);
+			resources.activeProjectAssets?.[ProjectAssetTypes.DATASETS].push(selectedItem);
+		}
+	});
+};
+
+const addAssetsToProject = async (projectName?: string) => {
+	if (props.selectedSearchItems.length === 0) return;
+
+	let projectId = '';
+	if (projectName !== undefined && typeof projectName === 'string') {
+		const project = projectsList.value.find((p) => p.name === projectName);
+		projectId = project?.id as string;
+	} else {
+		if (!validProject.value) return;
+		projectId = validProject.value.id;
+	}
+
+	addResourcesToProject(projectId);
+
+	emit('close');
+	router.push(`/projects/${projectId}`);
+};
+
+onMounted(async () => {
+	const all = await ProjectService.getAll();
```

Comment:
This is better to be called `projects` instead of `all`

Reply:
makes sense! I didnt even catch this had just moved the code I will fix that